forta-bots/Debt-Ceiling-Utilization/src/agent.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { Finding, HandleBlock, BlockEvent, keccak256, FindingSeverity, FindingType } from "forta-agent";
+import { MockEthersProvider, createAddress, TestBlockEvent } from "forta-agent-tools/lib/tests";
+import { provideHandleBlock } from "./agent";
+import { utils, BigNumber } from "ethers";
+import { createFinding } from "./utils";
+
+import { when, resetAllWhenMocks } from "jest-when";
+
+const TEST_THRESHOLD = BigNumber.from(1);
+
+describe("Debt Ceiling Utilization monitoring bot test suite", () => {
+  const mockProvider = new MockEthersProvider();
+
+  const mockNetworkManager = {
+    TeleportJoin: createAddress("0x5432"),
+    domain: keccak256("testDomain"),
+  };
+
+  const mockGetDebt = jest.fn();
+  const mockGetLine = jest.fn();
+
+  const mockFetcher = {
+    getDebt: mockGetDebt,
+    getLine: mockGetLine,
+  };
+
+  const handleBlock: HandleBlock = provideHandleBlock(mockNetworkManager as any, mockFetcher as any, TEST_THRESHOLD);
+
+  beforeEach(() => {
+    mockProvider.clear();
+    mockGetDebt.mockClear();
+    mockGetLine.mockClear();
+    resetAllWhenMocks();
+  });
+
+  it("should return no findings when debt/line is below threshold", async () => {
+    //line, debt, blockNumber
+    const TEST_CASES: [BigNumber, BigNumber, number][] = [
+      [BigNumber.from("9387592759532123"), BigNumber.from(234), 1234],
+      [BigNumber.from("19387592759532123"), BigNumber.from(55), 5678],
+      [BigNumber.from("92759532123"), BigNumber.from(-94595), 9999999],
```

Comment:
Is that negative `debt` intentional, does it makes sense that a debt can be negative?

Reply:
@e1Ru1o ,

Yes, according to the monitoring notion page "can be < 0 when settlement occurs before mint"